### ISSUE TITLE
metadata for OP, ED, PV etc.

### ISSUE THREAD

**Issue by papaj-na-wrotkach** [jellyfin/jellyfin-plugin-anidb#52]
I'd really love to have support for specials like openings and endings. @Nalsai in #6 said
> If you want any of these implemented or have another problem, make a new issue or message me, then we can work on it.
_Originally posted by @Nalsai in https://github.com/jellyfin/jellyfin-plugin-anidb/issues/6#issuecomment-1083036758_

and so here's the issue.

It could also be good idea to support use of anidb episode prefixes:
- regular episode: none
- special episode: `S`
- openings: `OP` (sometimes it can be sth like `OP1b`)
- endings: `ED` (sometimes it can be sth like `ED1b`)
- trailer/promo/ads: `T`

Alternatively the prefixes could also be:
- `E`, `EP` or `Episode ` - for regular episodes
- `SP` or `Special ` - for specials
- `PV`, `P`, `AD`, `PR` (not necessarily) or `Promo ` (not necessarily) - for trailer/promo/ads

There could be more types of episodes, but I haven't come across any.

And of course padded episode numbers (`EP1`, `EP01` or `EP0001`) should also be supported, but I guess it's already implemented.

**Comment by Nalsai** [comment]
I agree that support for that would be nice. Currently logic provided by the jellyfin server is used so only normal episodes are supported and specials (they internally have the season 0).

A custom filename parser would need to be implemented to support anidb episode prefixes. It should probably use the jellyfin internal parser as a base so existing supported names don't accidentally get broken.
Its not a priority for me, but I hope I have time to work on it.
